shopping-cart/test/com/commercetools/sunrise/shoppingcart/adddiscountcode/DefaultAddDiscountCodeControllerActionTest.java

```diff
@@ -0,0 +1,71 @@
+package com.commercetools.sunrise.shoppingcart.adddiscountcode;
+
+import com.commercetools.sunrise.framework.hooks.HookRunner;
+import io.sphere.sdk.carts.Cart;
+import io.sphere.sdk.carts.commands.CartUpdateCommand;
+import io.sphere.sdk.carts.commands.updateactions.AddDiscountCode;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.*;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link DefaultAddDiscountCodeControllerAction}.
+ */
+public class DefaultAddDiscountCodeControllerActionTest {
+    public static final String DISCOUNT_CODE = "SUNNY";
+    @Mock
+    private SphereClient sphereClient;
+    @Mock
+    private HookRunner hookRunner;
+    @Mock
+    private Cart cart;
+    @Captor
+    private ArgumentCaptor<CartUpdateCommand> cartUpdateCommandCaptor;
+
+    @InjectMocks
+    private DefaultAddDiscountCodeControllerAction defaultAddDiscountCodeControllerAction;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
```

Comment:
Do you like this option better than the class annotation?

Reply:
No, I prefer to use the annotation 😉 Looks like I copied it from `DefaultResetPasswordControllerActionTest` and would suggest to change that test to use annotations too. WDYT?

Reply:
That's what I thought, I wanted to ask you again just in case 😆 
Yep, we should change that test too then :)